### DISCUSSION TITLE
Citations in old format will break with "Oops! No text generated from Ollama, Please try again." error message.

### DISCUSSION THREAD

**Discussion by morgan55555** [open-webui/open-webui#7210] [Issues]
If your RAG collection was created on v0.4.2, citations will break on v0.4.3.
Output of model is working correctly, but instead of "References from" you will get error.
Recreation of RAG collection will help, but if collection is large enough, it will create some painful moments for users after update.
Update should not break existing collections.

After update on v0.4.3:
![image](https://github.com/user-attachments/assets/25d2d089-1818-42af-9384-8e52fc61636f)

And after collection recreation it was fixed:
![image](https://github.com/user-attachments/assets/2b3af371-926e-49f1-ba2a-18fc538a0848)

**Comment by tjbck** [comment]
Thanks for letting me know, investing.

**Comment by ahyaha** [comment]
I have the same issue

**Comment by tjbck** [comment]
Could you try the latest dev and check if that resolves the issue for you?

**Comment by ahyaha** [comment]
I am already using the latest update 4.0.3 and the problem still exist

Also, some ollama models are not replying at all

Inline citations also not showing

Waiting for your reply

**Reply by morgan55555** [reply]
If you are using default RAG prompt, you need to revert it to old version for ollama answers, in new RAG prompt models are getting strict instruction to skip RAG without source. And inline citations works on this feature too.

**Reply by morgan55555** [reply]
> I am already using the latest update 4.0.3 and the problem still exist 

And he asked you to use dev version, not 0.4.3, can you test it?

**Comment by ahyaha** [comment]
How to revert it back to the old version?

**Reply by morgan55555** [reply]
In RAG settings there is RAG prompt, you need to manually edit it.
(Edit new prompt, or just copy-paste old)

New RAG prompt is:

```
### Task:
Respond to the user query using the provided context, incorporating inline citations in the format [source_id] **only when the <source_id> tag is explicitly provided** in the context.

### Guidelines:
- If you don't know the answer, clearly state that.
- If uncertain, ask the user for clarification.
- Respond in the same language as the user's query.
- If the context is unreadable or of poor quality, inform the user and provide the best possible answer.
- If the answer isn't present in the context but you possess the knowledge, explain this to the user and provide the answer using your own understanding.
- **Only include inline citations using [source_id] when a <source_id> tag is explicitly provided in the context.**  
- Do not cite if the <source_id> tag is not provided in the context.  
- Do not use XML tags in your response.
- Ensure citations are concise and directly related to the information provided.

### Example of Citation:
If the user asks about a specific topic and the information is found in "whitepaper.pdf" with a provided <source_id>, the response should include the citation like so:  
* "According to the study, the proposed method increases efficiency by 20% [whitepaper.pdf]."
If no <source_id> is present, the response should omit the citation.

### Output:
Provide a clear and direct response to the user's query, including inline citations in the format [source_id] only when the <source_id> tag is present in the context.

<context>
{{CONTEXT}}
</context>

<user_query>
{{QUERY}}
</user_query>
```

Old RAG prompt:

```
You are given a user query, some textual context and rules, all inside xml tags. You have to answer the query based on the context while respecting the rules.

<context>
{{CONTEXT}}
</context>

<rules>
- If you don't know, just say so.
- If you are not sure, ask for clarification.
- Answer in the same language as the user query.
- If the context appears unreadable or of poor quality, tell the user then answer as best as you can.
- If the answer is not in the context but you think you know the answer, explain that to the user then answer with your own knowledge.
- Answer directly and without using xml tags.
</rules>

<user_query>
{{QUERY}}
</user_query>
```

**Comment by ahyaha** [comment]
The new one does not mention any resources even though I have 300 pdf files
in my knowledge.. shall i add them again to get the inline feature?

**Reply by morgan55555** [reply]
Yes, for proper work for inline citations it need to be readded. Seems that it need to fill a new fields in database. Also it will fix error with citations, it affects only old collections.

**Comment by ahyaha** [comment]
Can you share a photo of what the inline citations look like?

**Reply by morgan55555** [reply]
It just a filename with special formatting, like a button.
Pressing on it doesn't open any files or info, if you need open a link - you need to add own function with outlet filter, that changes this citations to a links.

Current inline citations (not clickable, you need to manually find corresponding document below):
![image](https://github.com/user-attachments/assets/19436a96-73a0-4bf1-8f86-87df0acb53fa)

Converted to links to the actual database by function:
![image](https://github.com/user-attachments/assets/44c5278e-2b5e-4301-bed0-2f520f5e569b)

**Comment by ahyaha** [comment]
File name or reference number?

**Reply by morgan55555** [reply]
It's filename, but you could make a reference number either, just by editing RAG prompt.
But with my LLM model I found that default citations with file names works better (more stable output), it depends on the model and on available VRAM.

**Comment by ahyaha** [comment]
Can you share with me please the RAG prompt that shows the reference
number?

**Reply by morgan55555** [reply]
Just try to edit RAG rules in prompt in natural language, it highly depends on the model that you using.

**Comment by ahyaha** [comment]
Another issue please, how to delete an ollama model .. before last update
we could do that but now i don't see the delete button

**Reply by morgan55555** [reply]
It moved in ollama connection menu in settings.

**Comment by ahyaha** [comment]
Another issue, it takes too long to load the knowledge base or modules tabs
.. before the update it was fast.. please fix

**Comment by ahyaha** [comment]
Feedback: I used the old RAG prompt but still the response doesn't show the
references like before the update .. why?

**Comment by ahyaha** [comment]
Waiting for your reply
